test/boost/s3_test.cc

```diff
@@ -664,6 +664,53 @@ SEASTAR_THREAD_TEST_CASE(test_download_data_source_proxy) {
     test_download_data_source(make_proxy_client, 3 * 1024);
 }
 
+SEASTAR_THREAD_TEST_CASE(test_object_copy) {
+    const sstring name(fmt::format("/{}/testobject-{}", tests::getenv_safe("S3_BUCKET_FOR_TEST"), ::getpid()));
+    const sstring name_copy(fmt::format("/{}/testobject-{}-copy", tests::getenv_safe("S3_BUCKET_FOR_TEST"), ::getpid()));
+
+    semaphore mem(16 << 20);
+    auto cln = make_minio_client(mem);
+    auto close_client = deferred_close(*cln);
+    constexpr std::string_view content{"1234567890"};
+
+    testlog.info("Put object {}", name);
+    temporary_buffer<char> data = sstring(content).release();
+    cln->put_object(name, std::move(data)).get();
+
+
+    testlog.info("Copy object {}", name);
+
+    cln->copy_object(name, name_copy).get();
+
+    size_t sz = cln->get_object_size(name_copy).get();
+    BOOST_REQUIRE_EQUAL(sz, content.size());
+
+    auto out = output_stream<char>(cln->make_upload_sink(name));
+
+    constexpr size_t chunk_size = 1_MiB;
+    constexpr size_t writes = 11;
+    auto rnd = tests::random::get_bytes(chunk_size);
+    for (unsigned ch = 0; ch < writes; ch++) {
+        out.write(reinterpret_cast<char*>(rnd.begin()), rnd.size()).get();
+    }
+
+    out.flush().get();
+    out.close().get();
+    cln->copy_object(name, name_copy, 5_MiB).get();
```

Comment:
What's the need to test how copy_object copies one object two times?

Reply:
first for PUT second for multipart

Reply:
Ah, you mean -- one object is "small", the other object is "large"? If so, plz, make it two test cases.